src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -750,20 +750,22 @@ private static object SplitWithPattern(ExecutionContext context, IScriptExtent e
                 separatorPattern = Regex.Escape(separatorPattern);
             }
 
-            if (limit < 0)
+            RegexOptions regexOptions = parseRegexOptions(options);
+
+            // if the limit is negative then set Regex to read from right to left
+            if (limit < 0) 
             {
-                // Regex only allows 0 to signify "no limit", whereas
-                // we allow any integer <= 0.
-                limit = 0;
+                regexOptions |= RegexOptions.RightToLeft;
+                limit *= -1;
```

Comment:
My opinion is that Codeacy's suggestion isn't ideal, but it might be worth sort of complying:

```csharp
int splitLimit = limit;
if (splitLimit < 0)
{
    regexOptions |= RegexOptions.RightToLeft;
    splitLimit *= -1;
}

// use splitLimit below
```